chatcommands.py

```diff
@@ -674,6 +674,28 @@ def master():
     os._exit(8)
 
 
+@command(privileged=True, aliases=['pip-update'])
+def pip_update():
+    """
+    Initiate `pip` requirement updates in userspace, or warn if we're in a venv.
+    :return: None
+    """
+    if not is_venv():
+        raise CmdException("Smokey is not running in a Python virtualenv.  Updating may cause other unrelated things "
+                           "to break.  Use pip-force to force the pip update anyways.")
+    else:
+        os._exit(20)
+
+
+@command(privileged=True, aliases=['pip-force'])
```

Comment:
Suggestion: add a `pip-force` alias to the `pip_update` command def, and check venv based on `alias_used` - our blacklist commands do this, if you want an example of how to use it.

Reply:
Working on this...